.github/pre-commit

```diff
@@ -8,6 +8,10 @@
 
 echo Shell: "${SHELL}"
 
+echo Updating auto-generated markdown files...
+make doc/gen-gomarkdoc
+git add ./doc
```

Comment:
I'm not sure that makes lots of sense ....  running `make...` is okay, but git add is redundant. Git will force you to add some uncommitted changes anyway. 

Also from another side not sure that we want to run it every commit.

Reply:
Yeah I mean it depends if we want to run it every commit. What about if we run it every release, then we could add it to the github actions pipeline and it would run for every release

Reply:
This, I just discussed with andrii about it and I will update the CI validation :+1:

Reply:
Done!